src/services/FavoriteProductsAPI.ts

```diff
@@ -0,0 +1,98 @@
+import { BASE_URL } from "@/Constants";
+import { Product } from "./ProductAPI";
+
+async function fetchFavorites(): Promise<
+  | {
+      ok: true;
+      favorites: Product[];
+    }
+  | {
+      ok: false;
+      message: string;
+    }
+> {
+  let resp = await fetch(`${BASE_URL}/api/products/favorite-products`, {
+    cache: "no-store",
+  });
+  if (resp.ok) {
+    let { favorites }: { favorites: Product[] } = await resp.json();
+    return {
+      ok: true,
+      favorites: favorites,
+    };
+  } else {
+    let jsonResp = await resp.json();
+    return {
+      ok: false,
+      message: jsonResp.message ?? "Something went wrong. Please try again.",
+    };
+  }
+}
+
+async function postFavoriteProduct(productId: string): Promise<
+  | {
+      ok: true;
+      product: Product;
+      message: string;
+    }
+  | {
+      ok: false;
+      message: string;
+    }
+> {
+  let resp = await fetch(
+    `${BASE_URL}/api/products/favorite-products/${productId}`,
+    {
+      method: "PUT",
+      cache: "no-store",
+    }
+  );
+  if (resp.ok) {
+    let jsonResp = await resp.json();
+    return {
+      ok: true,
+      product: jsonResp,
+      message: jsonResp.message,
```

Comment:
This is weird. You're using the same `jsonResp` both on the `product` key and on the `message` key...